test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -1255,49 +1255,90 @@ TEST_P(ProxyFilterIntegrationTest, SubClusterWithUnknownDomain) {
 
 // Verify that removed all sub cluster when dfp cluster is removed/updated.
 TEST_P(ProxyFilterIntegrationTest, SubClusterReloadCluster) {
+  if (GetParam() == Network::Address::IpVersion::v6) {
+    // For IPv6, ensure we're using explicit loopback address rather than just "localhost"
+    dns_hostname_ = "[::1]";
```

Comment:
So this is a red flag for the upgrade - why does updating cares cause behavioral changes such that we have to update the tests?  If there are unavoidable changes we're going to need all of them documented in changelogs.
/wait-any

Reply:
+1 It is not clear to me what sort of impact this will make on existing users. 

c-ares has recently been undergoing significant changes that brought subtle breaking changes. I'm unsure if this indicated poor c-ares coverage or tests are simply "adjusted" without considering user impact.